pallets/deitos/src/types.rs

```diff
@@ -275,6 +275,9 @@ impl<T: pallet::Config> AgreementDetails<T> {
             .storage_mb_per_block
             .saturating_mul(BalanceOf::<T>::saturated_from(
                 installment_length.saturated_into::<u128>(),
+            ))
+            .saturating_mul(BalanceOf::<T>::saturated_from(
```

Comment:
I think an extra saturating multiplication might be a slight overkill. We can do just o normal one since size is `u64` 
```rust
let cost = CurrentPrices::<T>::get()
    .storage_mb_per_block
    .saturating_mul(BalanceOf::<T>::saturated_from(
        installment_length.saturated_into::<u128>() * self.storage as u128,
    ));
```